### PR TITLE
lazarsoma-FTS-2

### PR DESCRIPTION
**Pull request by lazarsoma**

### CODE REVIEW

File: src/main/java/com/greenfoxacademy/levelup/configuration/CustomAuthenticationFailureHandler.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+public class CustomAuthenticationFailureHandler implements AuthenticationFailureHandler {
+
+  @Override
+  public void onAuthenticationFailure(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse, AuthenticationException e) throws IOException, ServletException {
+    httpServletResponse.setStatus(HttpStatus.UNAUTHORIZED.value());
+
+    String jsonPayload = "{\"message\" : \"%s\", \"timestamp\" : \"%s\" }";
+    httpServletResponse.getOutputStream().println(String.format(jsonPayload, e.getMessage(), Calendar.getInstance().getTime()));
+  }
+}
```
Comment (rozanagy) [inline src/main/java/com/greenfoxacademy/levelup/configuration/CustomAuthenticationFailureHandler.java]: EL EOF (empty line, end of file)

File: src/main/java/com/greenfoxacademy/levelup/configuration/SecurityConfig.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            .and()
+            .formLogin()
+            .failureHandler(customAuthenticationFailureHandler());
+  }
+
+  @Bean
+  public AuthenticationFailureHandler customAuthenticationFailureHandler() {
+    return new CustomAuthenticationFailureHandler();
+  }
+}
```
Comment (rozanagy) [inline src/main/java/com/greenfoxacademy/levelup/configuration/SecurityConfig.java]: EL EOF <3

**Review by rozanagy — COMMENTED** [review]
pls do the needful

File: src/main/java/com/greenfoxacademy/levelup/configuration/CustomAuthenticationFailureHandler.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import java.io.IOException;
+import java.util.Calendar;
+
+public class CustomAuthenticationFailureHandler implements AuthenticationFailureHandler {
+
+  @Override
+  public void onAuthenticationFailure(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse, AuthenticationException e) throws IOException, ServletException {
+    httpServletResponse.setStatus(HttpStatus.UNAUTHORIZED.value());
+
+    String jsonPayload = "{\"message\" : \"%s\", \"timestamp\" : \"%s\" }";
```
Comment (drexeverest) [inline src/main/java/com/greenfoxacademy/levelup/configuration/CustomAuthenticationFailureHandler.java:19]: Timestamp was not requested by the user story, but it's okey

**Review by drexeverest — CHANGES_REQUESTED** [review]

**Review by drexeverest — DISMISSED** [review]

**Review by rozanagy — DISMISSED** [review]

**Review by rozanagy — APPROVED** [review]
ok

**Review by drexeverest — DISMISSED** [review]

**Review by rozanagy — DISMISSED** [review]

**Review by hulfyback — DISMISSED** [review]

**Review by rozanagy — APPROVED** [review]
ok

**Review by hulfyback — APPROVED** [review]

**Pull request merged**